ocaml/xapi/rbac_audit.ml

```diff
@@ -350,85 +313,57 @@ let rec sexpr_args_of __context name rpc_value action =
   else (* heuristic 2: print uuid/refs arguments in the xapi call *)
     match rpc_value with
     | Rpc.String value -> (
-        let name_uuid_ref = get_obj_of_ref value in
-        match name_uuid_ref with
-        | None ->
-            if Astring.String.is_prefix ~affix:Ref.ref_prefix value then
-              (* it's a ref, just not in the db cache *)
-              Some
-                (get_sexpr_arg name
-                   (get_db_namevalue __context name action value)
-                   "" value
-                )
-            else (* ignore values that are not a ref *)
-              None
-        | Some (_name_of_ref_value, uuid_of_ref_value, ref_of_ref_value) ->
-            let name_of_ref_value =
-              match _name_of_ref_value with
-              | None ->
-                  get_db_namevalue __context name action ref_of_ref_value
-              | Some "" ->
-                  get_db_namevalue __context name action ref_of_ref_value
-              | Some a ->
-                  a
-            in
+      match Ref_index.lookup value with
+      | None ->
```

Comment:
could use `when` and and an extra case to make it flatter.

Reply:
yeah, the code around here is smelly, checking whether value is a ref should be much cheaper than looking it up